remove panic & wrapping informational error

emplam27 opened the pull request:
<!--  Thanks for sending a pull request! -->

**What this PR does / why we need it**:

ref. https://github.com/uber-go/guide/blob/master/style.md#dont-panic

Remove `panic()` in server production
- Change function from using panic to return error and caller to decide how to handle it.
- Wrapping error more informationally

Remove panic method & Wrapping errors in CRDT logics
- Currently, if error occurred in CRDT logic, server was shutdown because of `panic()` method
- Removed `panic()` methods & Change all errors able to logging

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes #497 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

**Additional documentation**:

<!--
This section can be blank if this pull request does not require a release note.

Please use the following format for linking documentation:
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

emplam27 committed:
remove panic & wrapping informational error

hackerwins reviewed (changes requested):
Thanks for your contribution.

I think the interface has changed only for the function below:
  - validation.RegisterValidation
  - validation.RegisterTranslation
  - config.configPath
  - time.ParseDuration

Next, it would be nice to remove the panic that can occur in CRDT data structure.

emplam27 committed:
Remove panic method & Wrapping errors
- Currently, if error occurred in CRDT logic, server was shutdown because of `panic()` method
- Removed `panic()` methods & Change all errors able to logging

emplam27 committed:
Fix integration and bench test failed

hackerwins reviewed (commented):
Thanks for the second work. I leave a few comments.

### We need to check whether an actual error can occur

While removing panic, some interfaces have been changed to return an error. The changes may reduce the usability of interfaces, It is necessary to check whether an actual error can occur.

For example, there is a situation like below:

An error chain `Text.String > RGATreeSplitNode.String > TextValue.String` need not return an actual error.

```go
func (t *TextValue) String() (string, error) {
	return t.value, nil
}
```

hackerwins committed:
Fix typo

emplam27 committed:
Rollback panics in primitive and counter

emplam27 committed:
Rollback panics in document json

emplam27 committed:
Rollback panics in document json

hackerwins committed:
Merge branch 'main' into remove-panic
